samples/bluetooth/ipsp/src/main.c

```diff
@@ -156,69 +158,39 @@ static inline bool get_context(struct net_context **udp_recv6,
 	return true;
 }
 
-static struct net_pkt *build_reply_pkt(const char *name,
-				       struct net_context *context,
-				       struct net_pkt *pkt)
+static int build_reply(const char *name,
+		       struct net_pkt *pkt,
+		       u8_t *buf)
 {
-	struct net_pkt *reply_pkt;
-	struct net_buf *tmp;
-	int header_len, recv_len, reply_len;
-
-	LOG_DBG("%s received %d bytes", name, net_pkt_appdatalen(pkt));
-
-	reply_pkt = net_pkt_get_tx(context, K_FOREVER);
-
-	recv_len = net_pkt_get_len(pkt);
-
-	tmp = pkt->frags;
-	/* Remove frag link so original pkt can be unrefed */
-	pkt->frags = NULL;
-
-	/* First fragment will contain IP header so move the data
-	 * down in order to get rid of it.
-	 */
-	header_len = net_pkt_appdata(pkt) - tmp->data;
+	int reply_len = net_pkt_remaining_data(pkt);
 
-	/* After this pull, the tmp->data points directly to application
-	 * data.
-	 */
-	net_buf_pull(tmp, header_len);
+	LOG_DBG("%s received %d bytes", name, reply_len);
 
-	/* Add the entire chain into reply */
-	net_pkt_frag_add(reply_pkt, tmp);
+	net_pkt_read_new(pkt, buf, reply_len);
 
-	reply_len = net_pkt_get_len(reply_pkt);
+	LOG_DBG("Received %d bytes, sending %d bytes", reply_len, reply_len);
 
-	LOG_DBG("Received %d bytes, sending %d bytes",
-		recv_len - header_len, reply_len);
-
-	return reply_pkt;
+	return reply_len;
 }
 
 static inline void pkt_sent(struct net_context *context,
 			    int status,
 			    void *token,
 			    void *user_data)
 {
-	if (!status) {
-		LOG_DBG("Sent %d bytes", POINTER_TO_UINT(token));
+	if (status >= 0) {
+		LOG_DBG("Sent %d bytes", status);
```

Comment:
The current description for `net_context_send_cb_t` reads:

>  @param status Value is set to 0 if all data was sent ok, <0 if
there was an error sending data. >0 amount of data that was
sent when not all data was sent ok.

So, either code here incorrect, or the callback description is outdated, and then please update it.

Reply:
It's the documentation which is incorrect, that will be fixed in a separate PR. Please dismiss you review so this can be applied.

Reply:
done